src/config.rs

```diff
@@ -1,78 +1,54 @@
 use serde_json::Value;
 
-use crate::context::Context;
-use crate::error::{ErrorRecorder, ValidationError};
+use crate::error::ValidationError;
 use crate::format::FormatChecker;
 use crate::resolver::Resolver;
 use crate::schemas;
-use crate::validators;
 use crate::validators::Validator;
 
+/// A structure to hold configuration for a single validation run.
 pub struct Config<'a> {
     schema: &'a Value,
     resolver: Resolver<'a>,
     draft: &'a dyn schemas::Draft,
 }
 
 impl<'a> Config<'a> {
+    /// Get the validator object for the draft in use.
     pub fn get_validator(&self, key: &str) -> Option<Validator> {
         self.draft.get_validator(key)
     }
 
+    /// Get the string format checker for the draft in use.
     pub fn get_format_checker(&self, key: &str) -> Option<FormatChecker> {
         self.draft.get_format_checker(key)
     }
 
+    /// Get the draft number in use.
     pub fn get_draft_number(&self) -> u8 {
         self.draft.get_draft_number()
     }
 
+    /// Get the metaschema associated with the draft in use.
     pub fn get_metaschema(&self) -> &Value {
         self.draft.get_schema()
     }
 
-    pub fn validate(
-        &self,
-        instance: &Value,
-        schema: &Value,
-        errors: &mut dyn ErrorRecorder,
-        validate_schema: bool,
-    ) -> Option<()> {
-        if validate_schema {
-            let metaschema = self.get_metaschema();
-            validators::descend(
-                self,
-                schema,
-                metaschema,
-                &Context::new(),
-                &Context::new(),
-                &Context::new_from(metaschema),
-                errors,
-            );
-            if errors.has_errors() {
-                return None;
-            }
-        }
-
-        validators::descend(
-            self,
-            instance,
-            schema,
-            &Context::new(),
-            &Context::new(),
-            &Context::new_from(schema),
-            errors,
-        )
-    }
-
+    /// Get the resolver for the parsing context.
     pub fn get_resolver(&self) -> &Resolver<'a> {
         &self.resolver
     }
 
+    /// Get the schema currently being checked against.
     pub fn get_schema(&self) -> &Value {
         &self.schema
     }
 
+    /// Create a new Config object from a given schema.
+    ///
+    /// Will attempt to automatically determine which draft to use based on the
```

Comment:
It's the other way around, it will use the passed `draft` value and fallback to determine it from the schema itself.